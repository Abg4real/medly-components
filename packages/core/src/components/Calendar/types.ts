import { HTMLProps, Omit, WithThemeProp } from '@medly-components/utils';

export interface Props extends Omit<HTMLProps<HTMLDivElement>, 'onChange'> {
    date: Date;
    onChange: (date: Date) => void;
    maxYear?: number;
    minYear?: number;
}

export interface DateProps extends HTMLProps<HTMLButtonElement>, WithThemeProp {
    isSelected: boolean;
    isInActiveMonth: boolean;
}
