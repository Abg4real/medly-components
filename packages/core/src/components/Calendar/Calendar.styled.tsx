import { centerAligned, css, styled } from '@medly-components/utils';
import Text from '../Text';
import TextField from '../TextField';
import { DateProps } from './types';

export const Calendar = styled('div')`
    padding: 1.6rem;
    background: ${({ theme }) => theme.calendar.bgColor};
    box-shadow: 0 0.2rem 0.8rem ${({ theme }) => theme.calendar.shadowColor};
    border-radius: ${({ theme }) => theme.calendar.borderRadius};
    width: max-content;
    height: max-content;
`;
export const MonthNavigation = styled.button<{ disabled?: boolean }>`
    border: none;
    background-color: transparent;
    border-radius: 50%;
    padding: 0.8rem;
    outline: none;
    cursor: pointer;
    color: ${({ theme, disabled }) => theme.colors.grey[disabled ? '400' : '800']};
    transition: all 100ms ease-out;
    svg {
        fill: ${({ theme, disabled }) => theme.colors.grey[disabled ? '400' : '800']};
    }

    ${centerAligned()}

    &:not(:disabled):hover {
        background-color: ${({ theme }) => theme.colors.grey[50]};
    }
    &:not(:disabled):active {
        background-color: ${({ theme }) => theme.colors.grey[100]};
    }
`;
export const Header = styled.div`
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: space-between;
`;

export const MonthAndYearSelection = styled.div`
    display: flex;
    margin: 0.5rem 0;
    align-items: center;
    justify-content: space-around;

    ul {
        max-height: 20rem;
        overflow: auto;
    }

    ${TextField.Style} {
        min-width: unset;
        max-width: 9.5rem;
        & > div {
            height: 3.5rem;
            padding: 0 0 0 1rem;
        }
    }
`;

export const CalendarGrid = styled.div`
    display: grid;
    grid-template-columns: repeat(7, 4rem);
    grid-auto-rows: auto;

    button > ${Text.Style} {
        user-select: none;
        text-align: center;
        padding: 0.8rem 1rem;
        margin-bottom: 0.8rem;
    }
`;

const activeDate = () => css`
    span {
        font-weight: bold;
    }
    &,
    &:hover {
        background-color: ${({ theme }) => theme.datePicker.selectedDateBgColor};
        color: ${({ theme }) => theme.datePicker.selectedDateColor};
    }
`;
const currentDate = () => css`
    border-color: ${({ theme }) => theme.datePicker.currentDateBorderColor};
`;
const nonActiveMonthDate = () => css`
    color: ${({ theme }) => theme.datePicker.nonActiveMonthDateColor};
`;

export const Date = styled('button').attrs({ type: 'button' })<DateProps>`
    background-color: transparent;
    border: 0.2rem solid transparent;
    text-decoration: none;
    cursor: pointer;
    user-select: none;
    border-radius: ${({ theme }) => theme.datePicker.borderRadius};
    padding: 0;
    margin: 0.4rem;
    box-sizing: border-box;
    height: 3.2rem;

    &:focus {
        outline: none;
    }

    &:disabled {
        cursor: not-allowed;
        color: ${({ theme }) => theme.datePicker.nonActiveMonthDateColor};
    }

    &:hover {
        background-color: ${({ theme }) => theme.datePicker.hoveredDateColor};
    }

    ${({ isSelected }) => isSelected && activeDate()};
    ${({ isCurrentDate }) => isCurrentDate && currentDate()};
    ${({ isInActiveMonth }) => !isInActiveMonth && nonActiveMonthDate()};

    ${centerAligned()}
`;
