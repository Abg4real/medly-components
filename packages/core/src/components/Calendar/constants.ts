export const WEEK_DAYS = ['S', 'M', 'T', 'W', 'T', 'F', 'S'];
export const CALENDAR_MONTHS = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
