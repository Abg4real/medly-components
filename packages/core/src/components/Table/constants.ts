import { TableColumnConfig } from './types';

export const rowActionsColumnConfig: TableColumnConfig = {
    title: '',
    field: 'row-actions',
    formatter: 'row-actions',
    hidden: false,
    frozen: true
};

export const loadingBodyData = [{ id: 'loading1' }, { id: 'loading2' }, { id: 'loading3' }, { id: 'loading4' }, { id: 'loading5' }];
