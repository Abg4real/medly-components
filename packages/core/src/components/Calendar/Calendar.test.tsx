import { cleanup, fireEvent, render } from '@test-utils';
import React from 'react';
import { Calendar } from './Calendar';
import { SHORT_CALENDAR_MONTHS } from './constants';
import { getMonthAndYearFromDate } from './helper';

const getDateValues = (container: HTMLElement) => {
    const month = container.querySelector('#test-calendar-month-selector-button span').innerHTML;
    const year = container.querySelector('#test-calendar-year-selector-button span').innerHTML;
    return {
        month,
        year
    };
};

describe('Calendar Component', () => {
    afterAll(cleanup);

    it('should render given date', () => {
        const date = new Date(2020, 0, 1),
            { container } = render(<Calendar id="test-calendar" date={date} onChange={jest.fn()} />);

        expect(container).toMatchSnapshot();
    });

    it('should render properly with with error state', () => {
        const date = new Date(2020, 0, 1),
            { container } = render(<Calendar id="test-calendar" date={date} onChange={jest.fn()} isErrorPresent />);

        expect(container).toMatchSnapshot();
    });

    it('should render current month if date is null', () => {
        const { container } = render(<Calendar id="test-calendar" date={null} onChange={jest.fn()} />),
            { month, year } = getMonthAndYearFromDate(new Date());
        const { month: monthInDOM, year: yearInDOM } = getDateValues(container);
        expect(monthInDOM).toEqual(`${SHORT_CALENDAR_MONTHS[month]}`);
        expect(yearInDOM).toEqual(year.toString());
    });

    it('should render min selectable date if current month is greater than max date and date is null', async () => {
        const { container } = render(
                <Calendar
                    id="test-calendar"
                    date={null}
                    onChange={jest.fn()}
                    maxSelectableDate={new Date(1902, 11, 31)}
                    minSelectableDate={new Date(1901, 0, 1)}
                />
            ),
            { month, year } = getMonthAndYearFromDate(new Date(1901, 0, 1));
        const { month: monthInDOM, year: yearInDOM } = getDateValues(container);
        expect(monthInDOM).toEqual(`${SHORT_CALENDAR_MONTHS[month]}`);
        expect(yearInDOM).toEqual(year.toString());
    });

    it('should call onChange with expected date', () => {
        const mockOnChange = jest.fn(),
            dateToSelect = new Date(2021, 1, 1),
            { container, getByText, getByTitle } = render(
                <Calendar
                    id="test-calendar"
                    date={null}
                    onChange={mockOnChange}
                    minSelectableDate={new Date(2020, 0, 1)}
                    maxSelectableDate={new Date(2022, 2, 1)}
                />
            );

        fireEvent.click(container.querySelector('#test-calendar-month-selector-button'));
        fireEvent.click(getByText('Feb'));
        fireEvent.click(container.querySelector('#test-calendar-year-selector-button'));
        fireEvent.click(getByText('2021'));
        fireEvent.click(getByTitle(dateToSelect.toDateString()));
        expect(mockOnChange).toHaveBeenCalledWith(dateToSelect);
    });

    it('should render previous month on clicking left arrow when current month is other than Jan', () => {
        const date = new Date(2020, 1, 1),
            { container } = render(<Calendar id="test-calendar" date={date} onChange={jest.fn()} />);
        fireEvent.click(container.querySelector('.calendar-month-navigation-go-back'));
        const { month, year } = getDateValues(container);
        expect(month).toEqual('Jan');
        expect(year).toEqual('2020');
    });

    it('should render dec month on clicking left arrow when current month is Jan', () => {
        const date = new Date(2020, 0, 1),
            { container } = render(<Calendar id="test-calendar" date={date} onChange={jest.fn()} />);
        fireEvent.click(container.querySelector('.calendar-month-navigation-go-back'));
        const { month, year } = getDateValues(container);
        expect(month).toEqual('Dec');
        expect(year).toEqual('2019');
    });

    it('should render next month on clicking right arrow when current month is other than Dec', () => {
        const date = new Date(2020, 1, 1),
            { container } = render(<Calendar id="test-calendar" date={date} onChange={jest.fn()} />);
        fireEvent.click(container.querySelector('.calendar-month-navigation-go-forward'));
        const { month, year } = getDateValues(container);
        expect(month).toEqual('Mar');
        expect(year).toEqual('2020');
    });

    it('should render Jan month on clicking right arrow when current month is Dec', () => {
        const date = new Date(2020, 11, 1),
            { container } = render(<Calendar id="test-calendar" date={date} onChange={jest.fn()} />);
        fireEvent.click(container.querySelector('.calendar-month-navigation-go-forward'));
        const { month, year } = getDateValues(container);
        expect(month).toEqual('Jan');
        expect(year).toEqual('2021');
    });

    it('should render 29 days in feb month in leap year', () => {
        const date = new Date(2020, 1, 1),
            { container, getByTitle } = render(<Calendar id="test-calendar" date={date} onChange={jest.fn()} />);

        expect(container).toContainElement(getByTitle('Sat Feb 29 2020'));
    });

    it('should render 28 days in feb month in non leap year', () => {
        const date = new Date(2021, 1, 1),
            { queryByText } = render(<Calendar id="test-calendar" date={date} onChange={jest.fn()} />);

        expect(queryByText('29')).toBeNull();
    });

    it('should disable dates which are are out of range ', () => {
        const date = new Date(2020, 11, 15),
            { getByTitle, container } = render(
                <Calendar
                    id="test-calendar"
                    date={date}
                    onChange={jest.fn()}
                    minSelectableDate={new Date(2020, 11, 10)}
                    maxSelectableDate={new Date(2020, 11, 20)}
                />
            );

        expect(getByTitle('Wed Dec 09 2020')).toBeDisabled();
        expect(getByTitle('Mon Dec 21 2020')).toBeDisabled();
        expect(container.querySelector('.calendar-month-navigation-go-back')).toBeDisabled();
        expect(container.querySelector('.calendar-month-navigation-go-forward')).toBeDisabled();
        expect(getByTitle('Tue Dec 15 2020')).not.toBeDisabled();
    });

    it('should change month to first non disable month on changing year if selected month is disabled in the newly selected year', () => {
        const date = new Date(2020, 11, 15),
            { getByText, container } = render(
                <Calendar
                    id="test-calendar"
                    date={date}
                    onChange={jest.fn()}
                    minSelectableDate={new Date(2020, 11, 10)}
                    maxSelectableDate={new Date(2021, 3, 20)}
                />
            );
        fireEvent.click(container.querySelector('#test-calendar-year-selector-button'));
        fireEvent.click(getByText('2021'));
        const { month, year } = getDateValues(container);
        expect(month).toEqual('Jan');
        expect(year).toEqual('2021');
    });

    it('should non change month on changing year if that month is non disable in the newly selected year', () => {
        const date = new Date(2020, 11, 15),
            { getByText, container } = render(
                <Calendar
                    id="test-calendar"
                    date={date}
                    onChange={jest.fn()}
                    minSelectableDate={new Date(2020, 11, 10)}
                    maxSelectableDate={new Date(2021, 11, 20)}
                />
            );

        fireEvent.click(container.querySelector('#test-calendar-year-selector-button'));
        fireEvent.click(getByText('2021'));
        const { month, year } = getDateValues(container);
        expect(month).toEqual('Dec');
        expect(year).toEqual('2021');
    });
});
