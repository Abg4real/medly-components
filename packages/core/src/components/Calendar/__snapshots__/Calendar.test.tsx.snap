// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar Component should render given date 1`] = `
.c4 {
  margin: 0;
  color: inherit;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.6rem;
}

.c5 {
  margin: 0;
  color: inherit;
  font-size: 2.4rem;
  font-weight: 400;
  -webkit-letter-spacing: -0.04rem;
  -moz-letter-spacing: -0.04rem;
  -ms-letter-spacing: -0.04rem;
  letter-spacing: -0.04rem;
  line-height: 3.2rem;
}

.c19 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c15 {
  font-size: 2.4rem;
  cursor: inherit;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
}

.c15 * {
  fill-opacity: 1;
  fill: #607890;
}

.c2 {
  border: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  padding: 1rem 2.4rem 1.2rem 2.4rem;
  border-radius: 0.8rem;
  background: transparent;
  color: #005AEE;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:hover {
  cursor: pointer;
}

.c2:focus {
  outline: none;
}

.c2:disabled {
  cursor: not-allowed;
}

.c2 .c14 * {
  fill: #005AEE;
}

.c2:hover {
  color: #1A6BF0;
}

.c2:hover .c14 * {
  fill: #1A6BF0;
}

.c2:disabled {
  color: #b0bcc8;
}

.c2 .c14 + .c3 {
  margin-left: 0.8rem;
}

.c2 .c3 + .c14 {
  margin-left: 0.8rem;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  margin: 5px;
  overflow: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  box-sizing: border-box;
  background-color: #ffffff;
  padding: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  border: 1px solid #dfe4e9;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
}

.c13 {
  top: 0;
  left: 0;
  cursor: text;
  position: absolute;
  margin-left: 1.6rem;
  -webkit-transition: all 150ms cubic-bezier(0.4,0,0.2,1);
  transition: all 150ms cubic-bezier(0.4,0,0.2,1);
  font-size: 1.6rem;
  line-height: 2.6rem;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
  touch-action: manipulation;
  -webkit-transform: translate3d(0,58%,0);
  -ms-transform: translate3d(0,58%,0);
  transform: translate3d(0,58%,0);
}

.c17 {
  pointer-events: none;
  margin: 0;
  padding: 0;
  position: absolute;
  top: 2.8rem;
  right: 1.6rem;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c11 {
  width: 100%;
  color: #13181D;
  font-size: 1.6rem;
  line-height: 2.6rem;
  cursor: text;
  box-sizing: border-box;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  padding-top: 1.4rem;
  padding-right: 5.6rem;
  padding-left: 1.6rem;
  padding-bottom: 1.4rem;
  border-radius: 0.4rem 0.4rem 0 0;
  background-color: #eff2f4;
  border: none;
  border-bottom: 0.1rem solid #566c82;
}

.c11:focus {
  outline: none;
}

.c11:focus::-webkit-input-placeholder {
  opacity: 1;
}

.c11:focus::-moz-placeholder {
  opacity: 1;
}

.c11:focus:-ms-input-placeholder {
  opacity: 1;
}

.c11:focus::placeholder {
  opacity: 1;
}

.c11::-webkit-input-placeholder {
  opacity: 0;
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
}

.c11::-moz-placeholder {
  opacity: 0;
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
}

.c11:-ms-input-placeholder {
  opacity: 0;
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
}

.c11::placeholder {
  opacity: 0;
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
}

.c11:not(:placeholder-shown) + .c12,
.c11:focus + .c12 {
  -webkit-transform: translate3d(0,20%,0) scale(0.75);
  -ms-transform: translate3d(0,20%,0) scale(0.75);
  transform: translate3d(0,20%,0) scale(0.75);
}

.c11 ~ .c12 {
  color: #566c82;
}

.c11 ~ .sc-AykKI * {
  fill: #566c82;
}

.c11 ~ .sc-AykKG {
  color: #566c82;
}

.c11:not(:disabled):hover,
.c11:not(:disabled):focus {
  padding-bottom: 1.3rem;
  border-bottom-width: 0.2rem;
}

.c11:not(:disabled):hover {
  border-color: #566c82;
}

.c11:disabled {
  cursor: not-allowed;
  color: #435465;
  background-color: #F7F8F9;
  border-color: #C7D0D8;
}

.c11:disabled ~ .c12 {
  color: #b0bcc8;
}

.c11:disabled ~ .sc-AykKI * {
  fill: #b0bcc8;
}

.c11:disabled ~ .sc-AykKG {
  color: #b0bcc8;
}

.c11:disabled ~ * {
  cursor: not-allowed;
}

.c11:focus,
.c11:focus:hover {
  border-color: #005AEE;
  background-color: #e6effd;
}

.c11:focus ~ .c12,
.c11:focus:hover ~ .c12 {
  color: #005AEE;
}

.c11:focus ~ .sc-AykKI *,
.c11:focus:hover ~ .sc-AykKI * {
  fill: #005AEE;
}

.c11:focus::-webkit-input-placeholder,
.c11:focus:hover::-webkit-input-placeholder {
  color: rgba(0,90,238,.2);
}

.c11:focus::-moz-placeholder,
.c11:focus:hover::-moz-placeholder {
  color: rgba(0,90,238,.2);
}

.c11:focus:-ms-input-placeholder,
.c11:focus:hover:-ms-input-placeholder {
  color: rgba(0,90,238,.2);
}

.c11:focus::placeholder,
.c11:focus:hover::placeholder {
  color: rgba(0,90,238,.2);
}

.c10 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  min-width: -webkit-max-content;
  min-width: -moz-max-content;
  min-width: max-content;
  margin: 0.8rem 0;
}

.c8 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: -webkit-max-content;
  min-width: -moz-max-content;
  min-width: max-content;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  margin: 0.8rem 0.8rem 0.8rem 0;
}

.c8 .c9 {
  margin: 0;
}

.c8 .c9 input {
  cursor: pointer;
}

.c8 .c9 input ~ .c16 {
  -webkit-transition: -webkit-transform 200ms ease-out;
  -webkit-transition: transform 200ms ease-out;
  transition: transform 200ms ease-out;
  cursor: pointer;
  -webkit-transform: translateY(-50%) rotate(0deg);
  -ms-transform: translateY(-50%) rotate(0deg);
  transform: translateY(-50%) rotate(0deg);
}

.c8 .c9 input ~ .c16 * {
  fill: #566c82;
}

.c8 .c9 input ~ .c16:hover * {
  fill: #13181D;
}

.c8 .c9 input:hover ~ .c16 * {
  fill: #13181D;
}

.c8 .c9 input:focus ~ .c16 * {
  fill: #005AEE;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 5px 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c6 .c7 {
  min-width: unset;
  max-width: 10rem;
}

.c6 .c7 input {
  max-width: 10rem;
  padding: 0.5rem;
}

.c6 .c7 input:hover,
.c6 .c7 input:focus {
  padding-bottom: 4px;
}

.c6 .c7 input ~ .c16 {
  top: 1.85rem;
  right: 0.5rem;
}

.c18 {
  display: grid;
  grid-template: repeat(7,auto) / repeat(7,auto);
}

.c18 > * {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 8px 10px;
}

.c20 {
  background-color: transparent;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  border-radius: 5px;
  color: #b0bcc8;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c20:focus {
  outline: none;
}

.c20:disabled {
  cursor: not-allowed;
  color: #b0bcc8;
}

.c21 {
  background-color: transparent;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  border-radius: 5px;
  background-color: #F38037;
  color: #ffffff;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c21:focus {
  outline: none;
}

.c21:disabled {
  cursor: not-allowed;
  color: #b0bcc8;
}

.c22 {
  background-color: transparent;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  border-radius: 5px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c22:focus {
  outline: none;
}

.c22:disabled {
  cursor: not-allowed;
  color: #b0bcc8;
}

<div>
  <div
    class="c0"
    id="test-calendar"
  >
    <div
      class="c1"
    >
      <button
        class="c2"
        color="default"
        type="button"
      >
        <span
          class="c3 c4"
        >
          &lt;
        </span>
      </button>
      <h3
        class="c3 c5"
      >
        Jan 2020
      </h3>
      <button
        class="c2"
        color="default"
        type="button"
      >
        <span
          class="c3 c4"
        >
          &gt;
        </span>
      </button>
    </div>
    <div
      class="c6"
    >
      <div
        class="c7 c8"
      >
        <div
          class="c9 c10"
        >
          <input
            aria-describedby="test-calendar-month-selector-helper-text"
            autocomplete="off"
            class="c11"
            id="test-calendar-month-selector-input"
            placeholder="Month"
            type="text"
            value="Jan"
          />
          <label
            class="c12 c13"
            for="test-calendar-month-selector-input"
          />
          <svg
            class="c14 c15 c16 c17"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
              fill="#000"
            />
          </svg>
          
        </div>
      </div>
      <div
        class="c7 c8"
      >
        <div
          class="c9 c10"
        >
          <input
            aria-describedby="test-calendar-year-selector-helper-text"
            autocomplete="off"
            class="c11"
            id="test-calendar-year-selector-input"
            placeholder="Year"
            type="text"
            value="2020"
          />
          <label
            class="c12 c13"
            for="test-calendar-year-selector-input"
          />
          <svg
            class="c14 c15 c16 c17"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
              fill="#000"
            />
          </svg>
          
        </div>
      </div>
    </div>
    <div
      class="c18"
    >
      <span
        class="c3 c19"
      >
        Su
      </span>
      <span
        class="c3 c19"
      >
        Mo
      </span>
      <span
        class="c3 c19"
      >
        Tu
      </span>
      <span
        class="c3 c19"
      >
        We
      </span>
      <span
        class="c3 c19"
      >
        Th
      </span>
      <span
        class="c3 c19"
      >
        Fr
      </span>
      <span
        class="c3 c19"
      >
        Sa
      </span>
      <button
        class="c20"
        title="Sun Dec 29 2019"
        type="button"
      >
        <span
          class="c3 c19"
        >
          29
        </span>
      </button>
      <button
        class="c20"
        title="Mon Dec 30 2019"
        type="button"
      >
        <span
          class="c3 c19"
        >
          30
        </span>
      </button>
      <button
        class="c20"
        title="Tue Dec 31 2019"
        type="button"
      >
        <span
          class="c3 c19"
        >
          31
        </span>
      </button>
      <button
        class="c21"
        title="Wed Jan 01 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          1
        </span>
      </button>
      <button
        class="c22"
        title="Thu Jan 02 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          2
        </span>
      </button>
      <button
        class="c22"
        title="Fri Jan 03 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          3
        </span>
      </button>
      <button
        class="c22"
        title="Sat Jan 04 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          4
        </span>
      </button>
      <button
        class="c22"
        title="Sun Jan 05 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          5
        </span>
      </button>
      <button
        class="c22"
        title="Mon Jan 06 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          6
        </span>
      </button>
      <button
        class="c22"
        title="Tue Jan 07 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          7
        </span>
      </button>
      <button
        class="c22"
        title="Wed Jan 08 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          8
        </span>
      </button>
      <button
        class="c22"
        title="Thu Jan 09 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          9
        </span>
      </button>
      <button
        class="c22"
        title="Fri Jan 10 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          10
        </span>
      </button>
      <button
        class="c22"
        title="Sat Jan 11 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          11
        </span>
      </button>
      <button
        class="c22"
        title="Sun Jan 12 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          12
        </span>
      </button>
      <button
        class="c22"
        title="Mon Jan 13 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          13
        </span>
      </button>
      <button
        class="c22"
        title="Tue Jan 14 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          14
        </span>
      </button>
      <button
        class="c22"
        title="Wed Jan 15 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          15
        </span>
      </button>
      <button
        class="c22"
        title="Thu Jan 16 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          16
        </span>
      </button>
      <button
        class="c22"
        title="Fri Jan 17 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          17
        </span>
      </button>
      <button
        class="c22"
        title="Sat Jan 18 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          18
        </span>
      </button>
      <button
        class="c22"
        title="Sun Jan 19 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          19
        </span>
      </button>
      <button
        class="c22"
        title="Mon Jan 20 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          20
        </span>
      </button>
      <button
        class="c22"
        title="Tue Jan 21 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          21
        </span>
      </button>
      <button
        class="c22"
        title="Wed Jan 22 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          22
        </span>
      </button>
      <button
        class="c22"
        title="Thu Jan 23 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          23
        </span>
      </button>
      <button
        class="c22"
        title="Fri Jan 24 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          24
        </span>
      </button>
      <button
        class="c22"
        title="Sat Jan 25 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          25
        </span>
      </button>
      <button
        class="c22"
        title="Sun Jan 26 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          26
        </span>
      </button>
      <button
        class="c22"
        title="Mon Jan 27 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          27
        </span>
      </button>
      <button
        class="c22"
        title="Tue Jan 28 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          28
        </span>
      </button>
      <button
        class="c22"
        title="Wed Jan 29 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          29
        </span>
      </button>
      <button
        class="c22"
        title="Thu Jan 30 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          30
        </span>
      </button>
      <button
        class="c22"
        title="Fri Jan 31 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          31
        </span>
      </button>
      <button
        class="c20"
        title="Sat Feb 01 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          1
        </span>
      </button>
      <button
        class="c20"
        title="Sun Feb 02 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          2
        </span>
      </button>
      <button
        class="c20"
        title="Mon Feb 03 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          3
        </span>
      </button>
      <button
        class="c20"
        title="Tue Feb 04 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          4
        </span>
      </button>
      <button
        class="c20"
        title="Wed Feb 05 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          5
        </span>
      </button>
      <button
        class="c20"
        title="Thu Feb 06 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          6
        </span>
      </button>
      <button
        class="c20"
        title="Fri Feb 07 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          7
        </span>
      </button>
      <button
        class="c20"
        title="Sat Feb 08 2020"
        type="button"
      >
        <span
          class="c3 c19"
        >
          8
        </span>
      </button>
    </div>
  </div>
</div>
`;
